modules/core/src/main/java/org/apache/ignite/internal/table/criteria/SqlSerializer.java

```diff
@@ -0,0 +1,218 @@
+        var elements = expression.getElements();
+
+        String template;
+
+        if (operator == Operator.AND || operator == Operator.OR) {
+            var delimiter = operator == Operator.AND ? ") AND (" : ") OR (";
+
+            template = IntStream.range(0, elements.length)
+                    .mapToObj(i -> String.format("{%d}", i))
+                    .collect(Collectors.joining(delimiter, "(", ")"));
+        } else if (operator == Operator.IN || operator == Operator.NOT_IN) {
+            var prefix = operator == Operator.IN ? "{0} IN (" : "{0} NOT (";
+
+            template = IntStream.range(1, elements.length)
+                    .mapToObj(i -> String.format("{%d}", i))
+                    .collect(Collectors.joining(", ", prefix, ")"));
+        } else {
+            template = ELEMENT_TEMPLATES.get(operator);
+        }
+
+        int end = 0;
+        var matcher = Pattern.compile("\\{(\\d+)\\}").matcher(template);
+
+        while (matcher.find()) {
+            if (matcher.start() > end) {
+                append(template.substring(end, matcher.start()));
+            }
+
+            int index = Integer.parseInt(matcher.group(1));
+            elements[index].accept(this, context);
+
+            end = matcher.end();
+        }
+
+        if (end < template.length()) {
+            append(template.substring(end));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public <T> void visit(Criteria criteria, @Nullable Void context) {
+        criteria.accept(this, context);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        return builder.toString();
+    }
+
+    private SqlSerializer append(String str) {
+        builder.append(str);
+
+        return this;
+    }
+
+    /**
+     * Builder.
+     */
+    public static class Builder  {
+        @Nullable
+        private String tableName;
+
+        @Nullable
+        private Set<String> columnNames;
+
+        @Nullable
+        private Criteria where;
+
+        /**
+         * Sets the table name (the number of entries that will be sent to the cluster in one network call).
+         *
+         * @param tableName Table name.
+         * @return This builder instance.
+         */
+        public SqlSerializer.Builder tableName(String tableName) {
+            this.tableName = tableName;
+
+            return this;
+        }
+
+        /**
+         * Sets the valid table column names to prevent SQL injection.
+         *
+         * @param columnNames Acceptable columns names.
+         * @return This builder instance.
+         */
+        public SqlSerializer.Builder columns(Set<String> columnNames) {
+            this.columnNames = columnNames;
+
+            return this;
+        }
+
+        /**
+         * Set the given criteria.
+         *
+         * @param where where condition.
+         */
+        public SqlSerializer.Builder where(@Nullable Criteria where) {
```

Comment:
What does mean `null` ?

Reply:
null means return all entries in record view. Fixed javadoc.